### PR TITLE
removes the `/node_metadata` GET endpoint

### PR DESCRIPTION
**Pull request by KPrasch**

### CODE REVIEW

File: newsfragments/3410.removal.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Removes the /node_metadata endpoint
```
Comment (cygnusv) [inline newsfragments/3410.removal.rst:1]:
```suggestion
Removes the /node_metadata GET endpoint
```

File: nucypher/network/server.py [hunk 2]
```diff
@@ -131,31 +120,28 @@ def node_metadata_exchange():
...
+                bytes(response), headers=response_headers, status=HTTPStatus.OK
+            )
 
         if metadata_request.announce_nodes:
             for metadata in metadata_request.announce_nodes:
-                try:
-                    metadata.verify()
-                except Exception:
-                    # inconsistent metadata
-                    pass
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply nucypher/network/server.py]: Apologies - I did not mean to include this change in this PR.  Reverted.

File: nucypher/network/server.py [hunk 3]
```diff
@@ -109,19 +108,9 @@ def public_information():
...
-            # Learn when learned about
-            this_node.start_learning_loop()
-
-        # All known nodes + this node
-        response_bytes = this_node.bytestring_of_known_nodes()
-        return Response(response_bytes, headers=headers)
-
     @rest_app.route('/node_metadata', methods=["POST"])
     def node_metadata_exchange():
+        response_headers = {"Content-Type": "application/octet-stream"}
```
Comment (cygnusv) [inline nucypher/network/server.py:113]: This can be used in L132 and remove L126.

**Review by cygnusv — APPROVED** [review]
LGTM! Just a minor comment.

**Comment by derekpierre** [comment]
Addressed RFC about `response_headers` in #3428 - acf014cdf1edb2e184bf16fb177678297a233cf1.

**Pull request merged**